zcash_primitives/src/transaction/components/orchard.rs

```diff
@@ -291,6 +352,62 @@ pub fn write_orchard_bundle<W: Write, D: OrchardDomainCommon + WriteBurn<W>>(
     Ok(())
 }
 
+/// Writes an [`orchard::Bundle`] in the appropriate transaction format.
+#[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+pub fn write_v6_orchard_bundle<W: Write, D: OrchardDomainCommon + WriteBurn<W>>(
+    mut writer: W,
+    bundle: Option<&orchard::Bundle<Authorized, Amount, D>>,
+) -> io::Result<()> {
+    if let Some(bundle) = &bundle {
+        write_orchard_bundle_contents(writer, bundle, 0)?; // Currently timelimit is hardcoded to 0
+    } else {
+        CompactSize::write(&mut writer, 0)?;
+    }
+
```

Comment:
why not just
```rust
    if bundle.is_none() {
        return(CompactSize::write(&mut writer, 0));
    }
    
  ..content of write_orchard_bundle_contents without generalization
```

Reply:
yeah, that's what I've done in the end after removing D everywhere